Objects/dictobject.c

```diff
@@ -4197,6 +4525,23 @@ dictiter_iternextkey(PyObject *self)
     return NULL;
 }
 
+static PyObject*
+dictiter_iternextkey(PyObject *self)
+{
+    dictiterobject *di = (dictiterobject *)self;
+    PyDictObject *d = di->di_dict;
```

Comment:
We'll need a bit more work to make concurrent `next(it)` thread-safe on the same iterator. That's not quite as urgent, but let's not lose track of it.

Reply:
I actually have that ready to go I believe in what will become the PR for thread-safe iteration, just need to get other things landed first :)